### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: <link>

### Overview

Update readme to conform to the latest README updates.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- x ] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,71 @@
...
+## Integrations
+<!-- Information about particularly relevant interfaces, endpoints or libraries related to the charm. For example, peer relation endpoints required by other charms for integration.--> 
+* image: The image relation provides the OpenStack image ID to the GitHub runners.
+* cos-agent: The COS agent subordinate charm provides observability using the Canonical
+Observability Stack (COS).
+
+## Learn more
+* [Read more](https://charmhub.io/github-runner-image-builder) <!--Link to the charm's official documentation-->
+* [Developer documentation](https://github.com/canonical/github-runner-image-builder-operator/blob/main/CONTRIBUTING.md) <!--Link to any developer documentation-->
+
```
Comment (yanksyoon) [inline README.md:67]: Add troubleshooting: * [Matrix](https://matrix.to/#/!DYvOMMMjuXPZRJYHdy:ubuntu.com?via=ubuntu.com&via=matrix.org)

File: README.md [hunk 2]
```diff
@@ -1,30 +1,71 @@
...
+After having deployed and integrated the charm with the GitHub runner charm, the image should start
+to build automatically and be provided to the GitHub runner automatically. The whole process takes
+around 10 minutes.
+
+## Integrations
+<!-- Information about particularly relevant interfaces, endpoints or libraries related to the charm. For example, peer relation endpoints required by other charms for integration.--> 
+* image: The image relation provides the OpenStack image ID to the GitHub runners.
+* cos-agent: The COS agent subordinate charm provides observability using the Canonical
+Observability Stack (COS).
+
```
Comment (yanksyoon) [inline README.md:63]: For a full list... link to charmhub integrations tab

File: README.md [hunk 3]
```diff
@@ -1,30 +1,71 @@
...
+Provide GitHub runner workload embedded snapshot image to the 
+[GitHub runner](https://charmhub.io/github-runner) charm. This charm is deployed as a VM and works
+on top of OpenStack infrastructure.
 
-## Description
+Like any Juju charm, this charm supports one-line deployment, configuration, integration, scaling,
+and more. For Charmed GitHub runner image builder, this includes support for configuring:
+* multi-arch
+* multi Ubuntu bases
+* Juju/Microk8s snap channels
```
Comment (yanksyoon) [inline README.md:12]: Capitalize microK8s (the K)

File: README.md [hunk 4]
```diff
@@ -1,30 +1,71 @@
-[![CharmHub Badge](https://charmhub.io/github-runner-image-builder/badge.svg)](https://charmhub.io/github-runner-image-builder)
-[![Promote charm](https://github.com/canonical/github-runner-operator/actions/workflows/promote_charm.yaml/badge.svg)](https://github.com/canonical/github-runner-operator/actions/workflows/promote_charm.yaml)
-[![Discourse Status](https://img.shields.io/discourse/status?server=https%3A%2F%2Fdiscourse.charmhub.io&style=flat&label=CharmHub%20Discourse)](https://discourse.charmhub.io)
+# GitHub runner image builder operator
+<!-- Use this space for badges -->
 
-# GitHub runner image builder
+Provide GitHub runner workload embedded snapshot image to the
```
Comment (erinecon) [inline README.md:4]:
```suggestion
A Juju charm that provides the GitHub runner workload embedded snapshot image to the 
```

File: README.md [hunk 5]
```diff
@@ -1,30 +1,71 @@
...
+[GitHub runner](https://charmhub.io/github-runner) charm. This charm is deployed as a VM and works
+on top of OpenStack infrastructure.
 
-## Description
+Like any Juju charm, this charm supports one-line deployment, configuration, integration, scaling,
+and more. For Charmed GitHub runner image builder, this includes support for configuring:
+* multi-arch
+* multi Ubuntu bases
+* Juju/Microk8s snap channels
+* external scripts
```
Comment (erinecon) [inline README.md:13]: Nitpick: I would capitalize the first word of each item in the list, for instance:
```
* Multi-arch
* Multi Ubuntu bases
etc.
```

File: README.md [hunk 6]
```diff
@@ -1,30 +1,71 @@
...
+[CharmHub Documentation](https://charmhub.io/github-runner-image-builder).
 
-## Development
+## Get started
+<!--Briefly summarize what the user will achieve in this guide.-->
+Deploy GitHub runner image builder with GitHub runners.
 
-This charm uses black and flake8 for formatting. Both run with the lint stage of tox.
+<!--Indicate software and hardware prerequisites-->
+You'll require a working [OpenStack installation](https://microstack.run/docs/single-node) with
```
Comment (erinecon) [inline README.md:23]:
```suggestion
You'll need a working [OpenStack installation](https://microstack.run/docs/single-node) with
```

File: README.md [hunk 7]
```diff
@@ -1,30 +1,71 @@
...
+flavors with minimum 2 CPU cores, 16GB RAM and 20GB disk.
```
Comment (erinecon) [inline README.md:24]:
```suggestion
flavors with a minimum of 2 CPU cores, 16GB RAM and 20GB disk.
```

File: README.md [hunk 8]
```diff
@@ -1,30 +1,71 @@
...
+--config openstack-user-name=<OPENSTACK-USER-NAME>
+
+juju integrate github-runner-image-builder github-runner
+```
+
+### Basic operations
+<!--Brief walkthrough of performing standard configurations or operations-->
+
+After having deployed and integrated the charm with the GitHub runner charm, the image should start
+to build automatically and be provided to the GitHub runner automatically. The whole process takes
```
Comment (erinecon) [inline README.md:55]:
```suggestion
to build and be provided to the GitHub runner automatically. The whole process takes
```
(Nitpick: To try and remove one "automatically" from the sentence.)

**Review by erinecon — COMMENTED** [review]
Thank you for updating the README! You've added a few comments based on our in-person review, and I've added a few more comments (mostly nitpicks).

File: README.md (hunk 2 above)
Comment (erinecon) [reply README.md:63]: I would rephrase slightly:
`For a full list of integrations, please refer to the [Charmhub documentation](https://charmhub.io/github-runner-image-builder/integrations).`

**Review by erinecon — COMMENTED** [review]
Thanks for updating! I have one more comment

**Review by erinecon — APPROVED** [review]
Thanks for updating the README!

File: README.md [hunk 9]
```diff
@@ -1,30 +1,74 @@
...
+* image: The image relation provides the OpenStack image ID to the GitHub runners.
+* cos-agent: The COS agent subordinate charm provides observability using the Canonical
+Observability Stack (COS).
+
+For a full list of integrations, please refer to the [Charmhub documentation](https://charmhub.io/github-runner-image-builder/integrations).
+
+## Learn more
+* [Read more](https://charmhub.io/github-runner-image-builder) <!--Link to the charm's official documentation-->
+* [Developer documentation](https://github.com/canonical/github-runner-image-builder-operator/blob/main/CONTRIBUTING.md) <!--Link to any developer documentation-->
+* [Troubleshooting](https://matrix.to/#/!DYvOMMMjuXPZRJYHdy:ubuntu.com?via=ubuntu.com&via=matrix.org)
```
Comment (amandahla) [inline README.md:69]: Is it expected? Troubleshooting links to the Matrix room.

Comment (erinecon) [reply README.md:69]: Yes, this is what I've been telling people to link to if there's no official troubleshooting documentation/guide.

**Review by amandahla — APPROVED** [review]

File: README.md (hunk 9 above)
Comment (erinecon) [reply README.md:69]: Varshi pointed out on another PR that this is the wrong Matrix room, please update to https://matrix.to/#/#charmhub-charmdev:ubuntu.com
